site/docs/plugins/menu.md

```diff
@@ -93,57 +93,373 @@ bot.start();
+We call this part of the menu a _dynamic range_.
+The easiest way to create a dynamic range is by using the `MenuRange` class that this plugin provides.
+A `MenuRange` provides you with exactly the same functions as a menu, but it does not have an identifier, and it cannot be registered.
+
+```ts
+function getRandomInt(min: number, max: number) {
+  return min + Math.floor(Math.random() * (max - min));
+}
+
+// Create a menu with a random number of buttons
+const menu = new Menu("random");
+
+menu.dynamic((_ctx) => {
+  const range = new MenuRange();
+  const buttonCount = getRandomInt(2, 10);
+  for (let i = 0; i < buttonCount; i++) {
+    range
+      .text(i.toString(), (ctx) => ctx.reply(`${i} selected`))
+      .row();
+  }
+  return range;
+});
+
+menu.text("Generate new", (ctx) => ctx.menu.update());
+```
+
+The range builder function that you pass to `dynamic` may be `async`, so you can even perform API calls or do database communication.
+
+Moreover, the range builder function takes a context object as a first argument.
+(This is unused in the example above).
+
+Optionally, as a second argument after `ctx`, you can receive a fresh instance of `MenuRange`.
+You can modify it instead of returning your own instance if that's what you prefer.
+
+```ts
+menu.dynamic((ctx, range) => {
+  for (const text of ["foo", "bar", "baz"]) {
+    range // no need for `new MenuRange()` or a `return`
+      .text(text, (ctx) => ctx.reply(text))
+      .row();
+  }
+});
+```
+
+## Answering Callback Queries Manually
+
+The menu plugin will call `answerCallbackQuery` automatically for its own buttons.
+You can set `autoAnswer: false` if you want to disable this.
 
-TODO: how does it work, what the hell happens under the hood, code examples
+```ts
+const menu = new Menu("id", { autoAnswer: false });
+```
+
+You will now have to call `answerCallbackQuery` yourself.
+This allows you to pass custom messages that are displayed to the user.
 
 ## Outdated Menus and Fingerprints
 
+Let's say you have a menu where a user can toggle notifications on and off, such as in the example [up here](#dynamic-labels).
+Now, if a user sends e.g. `/settings` twice, they will get the same menu twice.
+But changing the notification setting on one of the two messages will not update the other!
+
+It is clear that we cannot keep track of all settings messages in a chat, and update all old menus in the entire chat history.
+You would have to use so many API calls for this that Telegram will restrict your bot.
+You also need to buy a lot of storage to remember all the message identifiers of all menus in all chats.
+That is not practical.
+
+The solution to this is to check if a menu is outdated before performing any actions.
+That means that we will only update old menus if the user actually starts clicking any buttons on them.
+The menu plugin does this automatically for you, so you don't need to worry about it.
+
+You can configure what happens when an outdated menu is detected.
```

Comment:
Since there is full control, and not a choice between presets.
```suggestion
You can configure exactly what happens when an outdated menu is detected.
```